opentech/apply/funds/models.py

```diff
@@ -143,12 +158,60 @@ def process_form_submission(self, form):
                 response = cleaned_data.pop(field.id)
                 cleaned_data[field.block.name] = response
 
+        if cleaned_data.get('email'):
+            # TODO: link with active submission user.
+            User = get_user_model()
+            email = cleaned_data.get('email')
+            full_name = cleaned_data.get('full_name')
+
+            try:
+                user_data = {
+                    'username': full_name if full_name else email,
+                    'email': email,
+                    'is_active': False,  # User needs to activate account
+                }
+
+                if full_name and hasattr(User, 'full_name'):
+                    user_data['full_name'] = full_name
+
+                user = User.objects.create_user(**user_data)
```

Comment:
This should be the only part wrapped in the try except. 

```
try:
    user = User.objects.create_user(**user_data)
except IntegrityError:
    user = User.objects.get(email=email)
else:
    # send the activation
```

I think I would be happy to move the `send_activation_email` as part of our `create_user` method on the custom User.

Can you think of a reason not to?

Reply:
the only reason against it is that the only use case for this is when the user is created through submission

Reply:
I'm not sure i understand. You should still tighten up the try/except even if the except doesnt set the user.